src/outbox/outboxActions.js

```diff
@@ -46,23 +47,57 @@ export const messageSendComplete = (localMessageId: number): Action => ({
   local_message_id: localMessageId,
 });
 
-export const trySendMessages = (dispatch: Dispatch, getState: GetState): boolean => {
+export const trySendMessages = async (dispatch: Dispatch, getState: GetState): Promise<boolean> => {
   const state = getState();
   const auth = getAuth(state);
   const outboxToSend = state.outbox.filter(outbox => !outbox.isSent);
+  const oneWeekAgo = Date.now() / 1000 - 60 * 60 * 24 * 7;
```

Comment:
It always makes me a little nervous to see unit-conversion factors like this `/ 1000` appear silently -- it's too easy to compare one thing that's secretly in seconds to another that's secretly in milliseconds, and wind up like that [muffed Mars meteorology mission](https://www.wired.com/2010/11/1110mars-climate-observer-report/).

If we were writing in Haskell, I'd encode this in the type system, with a `newtype` that was an instance of the numeric typeclasses. In Rust or C++ I'd try to do something equivalent.

In the type system we have, though, I don't know of a good way to do that. So what we can't say with types, we say with names: I've often suffixed time variables with `Ms` to indicate they're in milliseconds.

For seconds, we could say `oneWeekAgoSec`. Or `oneWeekAgoTimestamp`, and declare that "timestamp" for us means "Unix seconds-since-epoch" -- since that's what it means in the `timestamp` property on `Message` which we get straight from the Zulip API.